lib/OpenQA/Utils.pm

```diff
@@ -641,6 +652,14 @@ sub render_escaped_refs {
     return bugref_to_href(url_to_href(xml_escape($text)));
 }
 
+sub find_bug_number {
+    my ($text) = @_;
+    if ($text =~ /\S+\-((?:$MARKER_REFS)\d+)\-\S+/) {
+        return $1;
+    }
+    return undef;
+}
```

Comment:
I think this can be written as simply

```perl
sub find_bug_number {
    my ($text) = @_;
    return $text =~ /\S+\-((?:$MARKER_REFS)\d+)\-\S+/;
}
```

Reply:
I try your code, but the unit test (t/16-utils.t) failed. it return value is `1`.  However, I write the same code in my test script, it works well. I did not know why.

Reply:
Because you got the number of times it matched. Something like `return $text =~ /\S+\-((?:$MARKER_REFS)\d+)\-\S+/ ? $1 : undef;` should work better.

Reply:
The original version looks fine to me.
But that's my personal preference.
I don't like to shorten things too much.
It can make debugging harder.

Reply:
Thanks for all the comments. I modified it as @kraih 's suggestion. It works well.